redis/conn.go

```diff
@@ -238,10 +257,26 @@ func DialContext(ctx context.Context, network, address string, options ...DialOp
 		}
 
 		tlsConn := tls.Client(netConn, tlsConfig)
-		if err := tlsConn.Handshake(); err != nil {
-			netConn.Close()
+
+		errc := make(chan error, 2)
```

Comment:
Why is this channel buffered if you're never actually consuming both events? Since both the timer's function / handshake are being performed in a goroutine, isn't it safe to just use an unbuffered channel?

Reply:
On second thought, I am guessing this is to avoid a leak where the goroutine is blocked trying to send to the channel?

Reply:
correct